.github/workflows/microbenchmark.yml

```diff
@@ -0,0 +1,51 @@
+name: microbenchmark
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - 'feature/**'
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+
+# limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+env:
+  GO_VERSION: 1.15.10
+
+jobs:
+  microbenchmark:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - id: buildkite
+        name: Run buildkite pipeline
+        uses: elastic/apm-pipeline-library/.github/actions/buildkite@current
+        with:
+          vaultUrl: ${{ secrets.VAULT_ADDR }}
+          vaultRoleId: ${{ secrets.VAULT_ROLE_ID }}
+          vaultSecretId: ${{ secrets.VAULT_SECRET_ID }}
+          pipeline: apm-agent-microbenchmark
+          waitFor: true
+          printBuildLogs: true
+          buildEnvVars: |
+            script=scripts/jenkins/bench.sh
+            repo=apm-agent-go
+            sha=feature/microbenchmark
+            GO_VERSION=${{ env.GO_VERSION }}
+            HOME=/var/lib/buildkite-agent
```

Comment:
For some reason Buildkite runners don't have a HOME hence the installation of go using `gimme` failed, see https://github.com/elastic/apm-agent-go/actions/runs/4873098627/jobs/8692189806?pr=1423#step:2:1369